### PR TITLE
[NP-4240] Get input update mode from context

### PR DESCRIPTION
**Pull request by KernelDeimos**
Allow `onKey` to be passed via context so developers don't need to override views on properties using `foam.u2.tag.Input`, which can be done incorrectly.

For example, the following override might be done:
```
{
  class: 'String',
  name: 'myProperty',
  updateVisibility: 'RO',
  view: {
    class: 'foam.u2.tag.Input',
    onKey: true
  }
}
```

Since the view has an override, `ModeAltView` will not be used unless the developer included it in their ViewSpec. In this case it's clear this is unintentional, since `updateVisibility` is set to `RO`.

### CODE REVIEW

File: src/foam/u2/Element.js [hunk 1]
```diff
@@ -861,7 +885,13 @@ foam.CLASS({
       class: 'Object',
       name: 'id',
       transient: true,
-      factory: function() { return this.NEXT_ID(); }
+      // factory: function() { return this.NEXT_ID(); }
+      factory: function() {
```
Comment (kgrgreer) [inline src/foam/u2/Element.js:889]: What is this for?

**Comment by kgrgreer** [comment]
What's the purpose of this?

File: src/foam/u2/Element.js (hunk 1 above)
Comment (KernelDeimos) [reply src/foam/u2/Element.js:889]: You know what this is for :P sorry, forgot to remove it

**Comment by kgrgreer** [comment]
I don't understand what this is for. Let's discuss.

**Comment by kgrgreer** [comment]
Have better method.

**Pull request closed without merging**